src/Std/Data/Iterators/Consumers/Access.lean

```diff
@@ -0,0 +1,52 @@
+/-
+Copyright (c) 2025 Lean FRO, LLC. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Paul Reichert
+-/
+prelude
+import Std.Data.Iterators.Consumers.Partial
+
+namespace Std.Iterators
+
+/--
+If possible, takes `n` steps with the iterator `it` and
+returns the `n`-th emitted value, or `none` if `it` finished
+before emitting `n` values.
+
+This function requires a `Productive` instance proving that the iterator will always emit a value
+after a finite number of skips. If the iterator is not productive or such an instance is not
+available, consider using `it.allowNontermination.toArray` instead of `it.toArray`. However, it is
```

Comment:
1. Presumably you didn't want `toArray` here.
2. I guess in this specific case you could unify the two functions using `partial_fixpoint`? Not sure if it's worth the trouble.
3. Not sure how much I love `seek`. Have you considered `atIdx?` like for tree maps?

Reply:
I fear that `atIdx?` might falsely suggest that this is an efficient lookup, but I agree that `seek` isn't nice as well. `List` would be the perfect role model, but unfortunately `List` doesn't have `seekIdx?`/`atIdx?` either.

Or how about some more verbose name like `atIdxSlow?`? Another alternative is to make this completely noncomputable if we don't expect this to be of any use for programming. The main use case for it is that it's useful for proofs.

I think I tend towards `atIdxSlow?`.

Reply:
`partial_fixpoint` won't work here either -- note that we need to assume `Productive` so that the function terminates.

Reply:
Works for me.